src/components/Chart.tsx

```diff
@@ -40,70 +39,83 @@ const DownloadButton = styled(Button)({
 })
 
 const Chart: React.FC<ChartProps> = ({ csvPath, title, width = '100%', height = '100%' }) => {
-    useEffect(() => {
-        const loadCSVData = async () => {
-            try {
-                const response = await axios.get(csvPath, { responseType: 'text' })
-                const csvText = response.data
-                return Papa.parse<string[]>(csvText, { header: false }).data
-            } catch (error) {
-                console.error('Error loading CSV data:', error)
-            }
+    const { data, error } = useCSVData(csvPath)
+
+    const formatNumber = (num: number) => {
+        if (num >= 1e9) {
+            return (num / 1e9).toFixed(1) + 'B'
+        } else if (num >= 1e6) {
+            return (num / 1e6).toFixed(1) + 'M'
+        } else if (num >= 1e3) {
+            return (num / 1e3).toFixed(1) + 'K'
+        } else {
+            return num.toString()
         }
+    }
+
+    useEffect(() => {
```

Comment:
Think this could be either a `useMemo` or a `useCallback`, but not a `useEffect`, because we are not dealing with React side effects here: https://www.w3schools.com/react/react_useeffect.asp

Reply:
I think this one needs to be followed up on too.